specifyweb/frontend/js_src/lib/components/DataModel/businessRuleDefs.ts

```diff
@@ -156,6 +157,18 @@ export const businessRuleDefs: MappedBusinessRuleDefs = {
           new tables.CollectingEvent.Resource()
         );
       }
+
+      // Set the default CoType
+      if (collectionObject.get('collectionObjectType') === undefined)
+        collectionObject
+          .rgetPromise('collection')
+          .then((collection: SpecifyResource<CollectionTable> | null) => {
+            if (collection !== null)
+              collectionObject.set(
+                'collectionObjectType',
+                collection.get('collectionObjectType')
+              );
```

Comment:
`CollectionObject -> Collection` shouldn't be null according to DataModel/types.ts but there are lots of tests where CO gets initialized as an empty resource and so it doesn't have a linked Collection/Cotype and so I changed the type here to allow null

Reply:
If this is something that is going to be fetched a lot, maybe it would be worth including the Collection -> CollectionObjectType in some context call? 

Specifically, maybe include the CollectionObjectType in the `/context/domain.json` endpoint? 

https://github.com/specify/specify7/blob/5c639bf75575e46df2b5a574269abe2e4a24fe81/specifyweb/context/views.py#L331-L345

https://github.com/specify/specify7/blob/5c639bf75575e46df2b5a574269abe2e4a24fe81/specifyweb/frontend/js_src/lib/components/DataModel/schema.ts#L83-L94

That way, if we can reasonably assume the schema endpoint has been fetched by the time the businessrule is being run, we can simply do something like: 

```ts
collectionObject.set(
  'collectionObjectType',
   schema.defaultCollectionObjectType
);

// maybe fallback to the promise if schema is undefined? 
```